packages/flutter/lib/src/services/binding.dart

```diff
@@ -214,6 +214,7 @@ class _DefaultBinaryMessenger extends BinaryMessenger {
       _handlers.remove(channel);
     else
       _handlers[channel] = handler;
+    drainChannelBuffers(channel);
```

Comment:
Like Ian said, if this is here, I'm also not sure what it should imply for MethodChannels. How do they get a chance to throw in the future? How could they control the caching per channel and/or per method? Maybe we can't entirely hide it and let the user instantiating the channel to also have some degree of control in those channel's own constructor?

Reply:
Another conceptual reason is that despite some variable name leaks here, a binary messenger really doesn't know what a 'channel' is, nor does its lifecycle match the cache keys. We're only calling the window.onPlatformMessage setter once but each channel is really flushing its own message from the queue (potentially with a bit of additional channel specific 'application logic').